proxy/src/telemetry/metrics/prometheus.rs

```diff
@@ -652,3 +620,26 @@ impl fmt::Display for Classification {
     }
 
 }
+
+
+// ===== impl Direction =====
+
+impl<'a> From<&'a ctx::Proxy> for Direction {
+    fn from(proxy: &'a ctx::Proxy) -> Self {
+        match proxy {
+            &ctx::Proxy::Inbound(_) => Direction::Inbound,
+            &ctx::Proxy::Outbound(_) => Direction::Outbound,
+        }
+    }
+}
+
+impl fmt::Display for Direction {
+
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
```

Comment:
nit: we should be consistent with regard to this whitespace.  i don't find much value in having whitespace between the parent and fn (though helpful definitely between fns, etc). here & throughout